src/tracks/Pattern.cpp

```diff
@@ -752,8 +752,6 @@ PatternView::~PatternView()
 
 void PatternView::update()
 {
-	m_pat->changeLength( m_pat->length() );
```

Comment:
[funkyEnding.mmp.zip](https://github.com/LMMS/lmms/files/455435/funkyEnding.mmp.zip)

![funkyending](https://cloud.githubusercontent.com/assets/6368949/18254418/d5ea6824-739d-11e6-9b43-33de6e5e4e2c.png)


![funkyendingreloaded](https://cloud.githubusercontent.com/assets/6368949/18254424/e3df819e-739d-11e6-8120-3597e12abb89.png)

Method to reproduce.
1. In a new project make a beat with a note at step 16 and hit play.
2. Right click and open sequence in the Piano Roll. Shift the note one 32nd step to the right.
It now still plays back as a one bar beat if you play from the BBEditor but if you now stop and play from the Piano Roll it will play back two bars, the second silent.
3. In the Song Editor, enter a BB track and drag it out. In my case this will make a stretch of one bar BB Tracks.
4. Save and reload. Now the BB tracks in the Song Editor are two bars.